homeassistant/components/broadlink/remote.py

```diff
@@ -282,9 +283,16 @@ async def async_learn_command(self, **kwargs):
         service = f"{RM_DOMAIN}.{SERVICE_LEARN_COMMAND}"
 
         if not self._attr_is_on:
-            _LOGGER.warning(
-                "%s canceled: %s entity is turned off", service, self.entity_id
+            error_message = f"{service} canceled: {self.entity_id} entity is turned off"
+
+            self.hass.bus.async_fire(
+                RM_EVENT_LEARNED_COMMAND,
```

Comment:
Will it be easier to build automations around this if the error cases have their own event type:
```suggestion
                RM_EVENT_LEARN_COMMAND_FAILED,
```

Reply:
Good idea. Have updated with the new failed event.